integration_tests/src/main/python/collection_ops_test.py

```diff
@@ -129,6 +129,11 @@ def test_size_of_map(data_gen, size_of_null):
             lambda spark: unary_op_df(spark, data_gen).selectExpr('size(a)'),
             conf={'spark.sql.legacy.sizeOfNull': size_of_null})
 
+@pytest.mark.parametrize('data_gen', [string_gen], ids=idfn)
+def test_reverse_strings(data_gen):
+    assert_gpu_and_cpu_are_equal_collect(
+            lambda spark: unary_op_df(spark, data_gen).selectExpr('reverse(a)'))
+
```

Comment:
I don't think this should be in collection ops test. Since this is testing for strings input, it should go to string test.

Reply:
The follow up work for `reverse(array)` then will go to array test.

Reply:
However, if you decided to combine both string and array tests then I'm fine with keeping it here.

Reply:
I am planning to combine the string and array tests since they are similar and we can reuse this test function. Thank you for the suggestion!